coresimd/wasm32/memory.rs

```diff
@@ -47,9 +48,10 @@ pub unsafe fn size(mem: i32) -> i32 {
 #[inline]
 #[cfg_attr(test, assert_instr("memory.grow", mem = 0))]
 #[rustc_args_required_const(0)]
-pub unsafe fn grow(mem: i32, delta: i32) -> i32 {
+#[stable(feature = "simd_wasm32", since = "1.33.0")]
+pub unsafe fn memory_grow(mem: u32, delta: usize) -> isize {
```

Comment:
`memory.grow` conceptually returns the value that `memory.size` would return after the resize, which suggests that it should be usize rather than isize (with the error value being `usize::MAX` rather than `-1`). However, I don't think it'll make a big difference in practice, so I don't feel strongly about it.

Reply:
Good point. User programs can use `as` to go either way anyways, and to me it feels that both approaches have downsides - using `isize` might mean that negative values might indicate success, while using `usize` will confuse those trying to use `-1` here to catch errors like the spec suggests =/

Reply:
That sounds reasonable to me! I've switched to `usize` and documented the return value on failure. Thanks for the suggestions @sunfishcode!